homework-03/utils/streams.js

```diff
@@ -0,0 +1,119 @@
+const commander = require('commander')
+const trough = require('through2')
+const fs = require('fs')
+const syspath = require('path')
+const csvjson = require('csvjson')
+
+function reverse() {
+  process.stdin.setEncoding('utf8')
+
+  process.stdin.on('readable', () => {
+    const chunk = process.stdin.read()
+    if (chunk !== null) {
+      process.stdout.write(`data: ${chunk}`)
+    }
+  })
+
+  process.stdin.on('end', () => {
+    process.stdout.write('end')
+  })
+}
+
+function transform() {
+  const transformStream = trough(write, end);
+  process.stdin.pipe(transformStream).pipe(process.stdout);
+}
+
+function write (buffer, encoding, next) {
+  this.push(buffer.toString().toUpperCase());
+  next();
+}
+
+function end (done) {
+  done();
+}
+
+function outputFile (filePath) {
+  if (fs.existsSync(filePath)) {
+    const stream = fs.createReadStream(filePath)
+    stream.pipe(process.stdout);
+  } else {
+    printError()
+  }
+}
+
+function convertFromFile(filePath) {
+  if (fs.existsSync(filePath) && filePath.endsWith('.csv')){
+    const read = fs.createReadStream(filePath)
+    const toObject = csvjson.stream.toObject()
+    const stringify = csvjson.stream.stringify()
```

Comment:
unfortunately, this doesn't handle the case when csv file is really huge (for example, 20000 lines +): the json becomes invalid.

Reply:
So what should I use instead?

Reply:
It's not mandatory, but I would write my own csvjson stream parser. However, It's behavior not described in task, so this note is optional